components/board/genericlinux/board.go

```diff
@@ -70,6 +70,19 @@ func RegisterBoard(modelName string, gpioMappings map[int]GPIOBoardMapping, useP
 					spis[spiConf.Name] = &spiBus{bus: spiConf.BusSelect}
 				}
 			}
+
+			var i2cs map[string]board.I2C
+			if len(conf.I2Cs) != 0 {
```

Comment:
I feel a little silly checking this length (why not get rid of the if statement and have an array of length 0 sometimes?), but wrote it like this to be consistent with the SPI initialization above and AnalogReader initialization below.

Reply:
I'd like to remove it in a separate PR, which can come after this is merged.